cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TranslatorSuite.scala

```diff
@@ -0,0 +1,434 @@
+    val results = breakSentence.transform(textDf1)
+      .withColumn("sentLen", flatten(col("result.sentLen")))
+      .select("sentLen").collect()
+    val headStr = results.head.getSeq(0).mkString("\n")
+    assert(headStr === "25")
+  }
+
+  override def testObjects(): Seq[TestObject[BreakSentence]] =
+    Seq(new TestObject(breakSentence, textDf1))
+
+  override def reader: MLReadable[_] = BreakSentence
+}
+
+class DictionaryLookupSuite extends TransformerFuzzing[DictionaryLookup]
+  with TranslatorKey with Flaky with TranslatorUtils {
+
+  import spark.implicits._
+
+  lazy val dictDf: DataFrame = Seq(List("fly")).toDF("text")
+
+  lazy val dictionaryLookup: DictionaryLookup = new DictionaryLookup()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setFromLanguage("en")
+    .setToLanguage("es")
+    .setTextCol("text")
+    .setOutputCol("result")
+
+  test("Break sentence ") {
+    val results = dictionaryLookup.transform(dictDf)
+      .withColumn("translations", flatten(col("result.translations")))
+      .withColumn("normalizedTarget", col("translations.normalizedTarget"))
+      .select("normalizedTarget").collect()
+    val headStr = results.head.getSeq(0).mkString("\n")
+    assert(headStr === "volar\nmosca\noperan\npilotar\nmoscas\nmarcha")
+  }
+
+  override def testObjects(): Seq[TestObject[DictionaryLookup]] =
+    Seq(new TestObject(dictionaryLookup, dictDf))
+
+  override def reader: MLReadable[_] = DictionaryLookup
+}
+
+class DictionaryExamplesSuite extends TransformerFuzzing[DictionaryExamples]
+  with TranslatorKey with Flaky with TranslatorUtils {
+
+  import spark.implicits._
+
+  lazy val dictDf: DataFrame = Seq(List(("fly", "volar"))).toDF("textAndTranslation")
+
+  lazy val dictionaryExamples: DictionaryExamples = new DictionaryExamples()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setFromLanguage("en")
+    .setToLanguage("es")
+    .setTextAndTranslationCol("textAndTranslation")
+    .setOutputCol("result")
+
+  test("Dictionary Examples") {
+    val results = dictionaryExamples.transform(dictDf)
+      .withColumn("examples", flatten(col("result.examples")))
+      .withColumn("sourceTerm", col("examples.sourceTerm"))
+      .withColumn("targetTerm", col("examples.targetTerm"))
+      .select("sourceTerm", "targetTerm").collect()
+    assert(results.head.getSeq(0).head.toString === "fly")
+    assert(results.head.getSeq(1).head.toString === "volar")
+  }
+
+  override def testObjects(): Seq[TestObject[DictionaryExamples]] =
+    Seq(new TestObject(dictionaryExamples, dictDf))
+
+  override def reader: MLReadable[_] = DictionaryExamples
+}
+
+class DocumentTranslatorSuite extends TransformerFuzzing[DocumentTranslator]
+  with TranslatorKey with Flaky {
+
+  import spark.implicits._
+
+  // TODO: Replace all of those SAS urls after 2022-07-07
+  lazy val sourceUrl: String = "https://mmlspark.blob.core.windows.net/datasets?sp=rl&st=2021-07-06T06" +
```

Comment:
are you able to get away without SAS URLs? the datasets blob is public I believe. Also you might want to pull the root URL into a val to keep code DRY

Reply:
Emmm I tried but seems can't access it directly using '"https://mmlspark.blob.core.windows.net/datasets", it returns "error": {
        "code": "InvalidRequest",
        "message": "Cannot access source document location with the current permissions.",
        "target": "Operation",
        "innerError": {
            "code": "InvalidDocumentAccessLevel",
            "message": "Cannot access source document location with the current permissions."
        }
    },

Reply:
On folders it might not work, but it should work on files. Is this manually inspecting a folder and loading content from it, or is it taking it URL by URL?